src/StaticWebAssetsSdk/Tasks/Data/StaticWebAssetEndpoint.cs

```diff
@@ -141,6 +141,25 @@ public StaticWebAssetEndpointProperty[] EndpointProperties
 
     public static IEqualityComparer<StaticWebAssetEndpoint> RouteAndAssetComparer { get; } = new RouteAndAssetEqualityComparer();
 
+    internal static IDictionary<string, List<StaticWebAssetEndpoint>> ToAssetFileDictionary(ITaskItem[] candidateEndpoints)
+    {
+        var result = new Dictionary<string, List<StaticWebAssetEndpoint>>(candidateEndpoints.Length / 2);
```

Comment:
Why the `/ 2`? I'm guessing it's because we expect there to be multiple endpoints per asset file, but why `2` specifically? Do we know that will give us a conservative upper bound for the number of entries?

Reply:
Yes, this is because for a given file there is the fingerprinted endpoint and the non-fingerprinted endpoint. For compressed files there's also the versions that have the encoding selector, so when we are grouping them, we are going to have a few less than half of them.